docs/3.0/tutorials/scraping.mdx

```diff
@@ -0,0 +1,503 @@
+
+
+@flow(log_prints=True)
+def analyze_repo_health(repos: List[str]):
+    """Analyze issue health metrics for GitHub repositories"""
+    for repo in repos:
+        print(f"Analyzing {repo}...")
+        
+        # Fetch and analyze all issues
+        fetch_page_of_issues(repo)
+
+
+@task(log_prints=True)
+def fetch_page_of_issues(repo: str, page: int = 1) -> Optional[dict]:
+    """Fetch a page of issues for a GitHub repository"""
+    try:
+        response = httpx.get(
+            f"https://api.github.com/repos/{repo}/issues",
+            params={"page": page, "state": "all", "per_page": 100}
+        )
+        response.raise_for_status() # Raise an exception if the response is not a 2xx status code
+        return response.json()
+    except Exception as e:
+        print(f"Error fetching issues for {repo}: {e}")
+        return None
+
+
+if __name__ == "__main__":
+    analyze_repo_health([
+        "PrefectHQ/prefect",
+        "this-repo-does-not-exist/404" # This repo will trigger an error
+    ])
+```
+</Expandable>
+
+## Ingest large amounts of data
+
+Use pagination to fetch large amounts of data and run tasks in parallel to analyze the data efficiently:
+
+```python
+@flow(log_prints=True)
+def analyze_repo_health(repos: [str]):
+    """Analyze issue health metrics for GitHub repositories"""
+    all_issues = []
+
+    for repo in repos:
+        for page in range(1, 3):  # Get first 2 pages
+            issues = fetch_page_of_issues(repo, page)
+            if not issues:
+                break
+            all_issues.extend(issues)
+    
+    # Run issue analysis tasks in parallel
+    for issue in all_issues:
+        analyze_issue.submit(issue)
+    
+    # Wait for all analysis tasks to complete
+    for detail in issue_details:
+        result = detail.result()
+        print(f"Analyzed issue #{result['number']}")
+```
+
+<Expandable title="full example">
+Run the following code to see pagination and parallel tasks in action:
+
+```python
+from typing import List, Optional
+import httpx
+
+from prefect import flow, task
+
+
+@flow(log_prints=True)
+def analyze_repo_health(repos: List[str]):
+    """Analyze issue health metrics for GitHub repositories"""
+    for repo in repos:
+        print(f"Analyzing {repo}...")
+        
+        # Fetch and analyze all issues
+        fetch_repo_issues(repo)
+
+
+@flow
+def fetch_repo_issues(repo: str):
+    """Fetch all issues for a single repository"""
+    all_issues = []
+    page = 1
+    
+    for page in range(1, 3):  # Limit to 2 pages to avoid hitting rate limits
+        issues = fetch_page_of_issues(repo, page)
+        if not issues or len(issues) == 0:
+            break
+        all_issues.extend(issues)
+        page += 1
+
+    issue_details = []
+    for issue in all_issues[:5]:  # Limit to 5 issues to avoid hitting rate limits
+        issue_details.append(
+            fetch_issue_details.submit(repo, issue['number'])  # Make requests in parallel
```

Comment:
Either inline, or probably below, maybe call out what `.submit` is doing briefly and send folks to another docs page to learn more. It's kind of subtle as is.

Reply:
Same note where use ".submit" above in short code example.

Reply:
https://github.com/PrefectHQ/prefect/pull/15990/commits/f380af8a709379966080c0c7526392ecff53d88f

I'll augment the "Next steps" section with links to additional resources in a follow-up commit.

Reply:
This is subtle, but the example is technically doing concurrent execution rather than parallel execution (with Dask or Ray it would be true parallelism).

Reply:
Okay, I read through https://docs.prefect.io/v3/develop/task-runners more carefully and I think I understand the difference now. The difference is that `submit` uses `ThreadPoolTaskRunner` by default, which uses a thread pool and so the tasks aren't run in a true parallel fashion (distributed across multiple CPUs). cc @aaazzam 

I just pushed a commit which uses the correct language consistently throughout the new tutorials:
https://github.com/PrefectHQ/prefect/pull/15990/commits/9bc36621e4f55f5629dca9d819cd07de7c729b14